src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryFile.cs

```diff
@@ -0,0 +1,599 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    /// <summary>
+    /// TODO:
+    /// 1) Add class description.
+    /// 2) Logs to LogDebug or LogTrace.
+    /// 3) Make sure all meaningful log places are logging.
+    /// 4) long and int consistency - ensure > 2GB mmaps are handled.
+    /// 5) put all private methods at the end.
+    /// </summary>
+    public class SharedMemoryFile : IDisposable
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// <see cref="MemoryMappedFile"/> containg the header + content of this <see cref="SharedMemoryFile"/>.
+        /// </summary>
+        private readonly MemoryMappedFile _mmf;
+
+        /// <summary>
+        /// Name of the <see cref="MemoryMappedFile"/> backing this <see cref="SharedMemoryFile"/>.
+        /// </summary>
+        private readonly string _mapName;
+
+        public SharedMemoryFile(ILogger logger, string mapName, MemoryMappedFile mmf)
+        {
+            _logger = logger;
+            _mapName = mapName;
+            _mmf = mmf;
+        }
+
+        public static SharedMemoryFile Create(ILogger logger, string mapName, long size)
+        {
+            if (TryCreate(logger, mapName, size, out MemoryMappedFile mmf))
+            {
+                return new SharedMemoryFile(logger, mapName, mmf);
+            }
+            else
+            {
+                throw new Exception($"Cannot create MemoryMappedFile {mapName} with {size} bytes");
```

Comment:
do not throw new exception through out this file. Let the caller handle the error case.